ci/integration-tests-linux.groovy

```diff
@@ -1,68 +1,114 @@
-          credentialsId: 'ca159ad3-7323-4564-818c-46a8f03e1389',
-          variable: 'DCOS_TEST_LICENSE'],
-          [$class: 'UsernamePasswordMultiBinding',
-          credentialsId: '323df884-742b-4099-b8b7-d764e5eb9674',
-          usernameVariable: 'DCOS_USERNAME',
-          passwordVariable: 'DCOS_PASSWORD']
-        ]) {
-          unstash 'dcos-linux'
+        stage("Launch AWS Cluster") {
+            steps {
+                withCredentials(credentials) {
+                    script {
+                        master_ip = sh(script: '''docker run --rm -v $PWD:/usr/src -w /usr/src \
+                                      -v ${CLI_TEST_SSH_KEY_PATH}:${CLI_TEST_SSH_KEY_PATH} \
+                                      -e DCOS_TEST_INSTALLER_URL \
+                                      -e AWS_ACCESS_KEY_ID -e AWS_SECRET_ACCESS_KEY \
+                                      -e DCOS_USERNAME -e DCOS_PASSWORD \
+                                      -e DCOS_TEST_LICENSE -e CLI_TEST_SSH_KEY_PATH \
+                                      python:3.7-stretch bash -c " \
+                                        cd scripts; \
+                                        python3 -m venv env; \
+                                        source env/bin/activate; \
+                                        pip -q install --upgrade pip==18.1 setuptools; \
+                                        pip -q install -r requirements.txt; \
+                                        ./launch_aws_cluster.py create"''',
+                                returnStdout: true).trim()
+                    }
+                    stash includes: 'scripts/config.json', name: 'aws-config'
+                }
+            }
+        }
 
-          sh '''
-            docker run --rm -v $PWD:/usr/src -w /usr/src \
-              -v ${DCOS_TEST_SSH_KEY_PATH}:${DCOS_TEST_SSH_KEY_PATH} \
-              -e DCOS_TEST_INSTALLER_URL \
-              -e AWS_ACCESS_KEY_ID -e AWS_SECRET_ACCESS_KEY \
-              -e DCOS_USERNAME -e DCOS_PASSWORD \
-              -e DCOS_TEST_LICENSE -e DCOS_TEST_SSH_KEY_PATH \
-              python:3.7-stretch bash -exc " \
-                mkdir -p build/linux; \
-                make plugin; \
-                cd scripts; \
-                python3 -m venv env; \
-                source env/bin/activate; \
-                pip install --upgrade pip==18.1 setuptools; \
-                pip install -r requirements.txt; \
-                wget -O env/bin/dcos https://downloads.dcos.io/cli/testing/binaries/dcos/linux/x86-64/master/dcos; \
-                dcos cluster remove --all; \
-                ./run_integration_tests.py"
-          '''
+        stage("Run integration tests") {
+            agent { label 'py36' }
+            steps {
+                unstash "dcos-${os}"
+                withEnv(["DCOS_TEST_URL=${master_ip}", "OS=${os}"]) {
+                    withCredentials(credentials) {
+                        sh '''scripts/run_integration_tests.sh'''
```

Comment:
Regarding moving scripts from Python to Bash, I think in general it creates a bigger gap between UNIX and Windows Jenkinsfile / pipelines, as it makes it hard to share the same script. Even though we have winbash installed on windows nodes, Python is much more easier to run cross-platforms.

Reply:
Good point but previously we have `run_integration_tests` in Python and Windows was not using it. We can think about moving from bash to python to support windows but I'm not sure if that's cross platform since

1.  dcos binary has `.exe` extension on Windows
2. On Mac/Linux we use `make plugin` while on Windows we are using `make python`

Full diff:

```diff
0a1,2
> #!/usr/bin/env bash
> 
4,6c6
< mkdir -p build/windows
< make python
< python scripts/plugin/package_plugin.py
---
> DCOS_EXPERIMENTAL=1 make plugin
8,11c8,10
< make env
< rm -f ./env/Scripts/dcos.exe
< mv ../../../dcos.exe dist
< PATH=$PWD/dist:$PATH
---
> source env/bin/activate
> wget -qO env/bin/dcos https://downloads.dcos.io/cli/testing/binaries/dcos/${OS}/x86-64/master/dcos
> chmod +x env/bin/dcos
15,16c14
< ./env/Scripts/pytest -vv -x --durations=10 -p no:cacheprovider tests/integrations
< 
---
> py.test -vv -x --durations=10 -p no:cacheprovider tests/integrations
```

Reply:
closing as it's more of a general comment, indeed currently they are not fully shared yet